### PR TITLE
Added  PHP 8.2 support, dropped PHP 7 support

### PR DESCRIPTION
**Pull request by glo71317**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### CODE REVIEW

File: .laminas-ci.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
     "ignore_php_platform_requirements": {
-        "8.1": true
+        "8.2": true
```
Comment (Ocramius) [inline .laminas-ci.json]: Probably not needed anymore? 🤔

File: composer.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
     "extra": {
     },
     "require": {
-        "php": "^7.3 || ~8.0.0 || ~8.1.0",
+        "php": "^7.3 || ~8.0.0 || ~8.1.0 || ~8.2.0",
```
Comment (Ocramius) [inline composer.json]: We should drop PHP 7

Comment (Ocramius) [reply composer.json]: `config.platform` needs to pin the PHP version

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .laminas-ci.json (hunk 1 above)
Comment (glo71317) [reply .laminas-ci.json]: laminas/laminas-captcha has php8.2 support as this required php8.2 support

File: composer.json (hunk 2 above)
Comment (glo71317) [reply composer.json]: done

File: .laminas-ci.json (hunk 1 above)
Comment (Ocramius) [reply .laminas-ci.json]: Yes, but are current dependencies precluding a PHP 8 upgrade? This file introduces a CI error suppression/workaround, so it should be possible to remove it, if all dependencies are 8.2-ready

**Comment by Ocramius** [comment]
Hmm, CI failures highlights some needs for changes.

File: .laminas-ci.json (hunk 1 above)
Comment (glo71317) [reply .laminas-ci.json]: Yes all dependencies have 8.2 support

Comment (Ocramius) [reply .laminas-ci.json]: Then this file can be dropped 👍

Comment (glo71317) [reply .laminas-ci.json]: 👍

**Comment by Ocramius** [comment]
The test failures need to be addressed, not worked around 😁

**Comment by Ocramius** [comment]
I think the minimum PHPUnit version needs to be raised to 9.5 here

**Comment by glo82145** [comment]
> I think the minimum PHPUnit version needs to be raised to 9.5 here

@Ocramius Thank you so much , this solution has worked here. Still One test case is failing, looking into that

**Comment by glo82145** [comment]
@Ocramius there is one testcase failing due to squizlabs/php_codesniffer. It could be due to same reason as you mentioned in PR https://github.com/laminas/laminas-oauth/pull/20. Can you please confirm?

**Comment by Ocramius** [comment]
I wonder if we should just skip PHPCS: we don't want to improve CS on these components, as it poses more risk (regressions).

Alternatively, upgrading of `laminas/laminas-coding-standard` is possible, but with very conservative changes.

**Comment by glo82145** [comment]
> I wonder if we should just skip PHPCS: we don't want to improve CS on these components, as it poses more risk (regressions).
> 
> Alternatively, upgrading of `laminas/laminas-coding-standard` is possible, but with very conservative changes.

@Ocramius Yes, upgrading `laminas/laminas-coding-standard` would ask for lot of changes which will make things more complicated at this stage. Hence it should be investigated further to find appropriate solution.  But as it is blocking this current PR from merging and creating a blocker, Can we just skip this one  from this PR and take it up as different ticket for further investigation..

**Comment by Ocramius** [comment]
> But as it is blocking this current PR from merging and creating a blocker

There is no blocker over here? Simply means it won't be merged until green here. 

 > Can we just skip this one from this PR and take it up as different ticket for further investigation..

IMO no: merging broken stuff is just going to lead to more load on the maintainers: we don't leave stuff half-done here, as it's mostly making thing complicated for everyone else.

File: phpcs.xml [hunk 3]
```diff
@@ -17,24 +17,4 @@
     <file>test</file>
     <exclude-pattern>*/_files/*</exclude-pattern>
     <exclude-pattern>*/TestAsset/*</exclude-pattern>
-
-    <!-- Include all rules from Laminas Coding Standard -->
-    <rule ref="LaminasCodingStandard">
```
Comment (Ocramius) [inline phpcs.xml]: The `LaminasCodingStandard` should still be preserved

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by glo82145** [comment]
@Ocramius We have upgraded the laminas/laminas-coding-standard and updated the code as per the requirement.

**Review by Ocramius — APPROVED** [review]
I had to rewrite the patch to fix multiple changes done incorrectly, such as removed `@throws` descriptions, incorrect/wrong `@var` annotations and removed `@deprecated` comments.

I also squashed previous commits, which were just saying mentioning that some issues were being fixed, without any mention if which they were.

In practice, I understand that you were assigned this task (outsourced?), and that you had no experience with this, but I really wish you had put some thinking into the errors that the CS tooling reported, rather than just steamrolling over the pre-existing code.

Instead, it seems like you've bashed your keyboard until you reached green, which is a bit sad, since it just means double work for maintainers :-( 

Anyway, this is now ready for release.

**Pull request merged**